### PR TITLE
Support dynamic tables

### PR DESCRIPTION
**Pull request by twodayslate**

### CODE REVIEW

File: Example-iOS/ViewControllers/DynamicViewController.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          )
+          self.tableView.insertRows(at: [IndexPath(row: self.dynamicRows.count, section: 0)], with: .automatic)
+        })
+      ] + dynamicRows
+
+      return [
+        Section(title: "Tap Action", rows: rows as! [Row & RowStyle])
+      ]
+    }
+    set {}
```
[the remark this answers is not in the document]
Comment (twodayslate) [reply Example-iOS/ViewControllers/DynamicViewController.swift]: Can I ignore a violation?

**Comment by bcylin** [comment]
Hi @twodayslate, thanks for opening this pull request. Can you briefly explain what these changes are for?

**Comment by twodayslate** [comment]
@bcylin I included an example to demonstrate the capabilities. The implementation is rather simplistic but it allows for the use of [`insertRows`](https://developer.apple.com/documentation/uikit/uitableview/1614879-insertrows) and the like without having to redefine `tableContents`- all you have to do is just define a getter for `tableContents` and the table will be reloaded auto-magically. 

I would think this fixes the main limitation of `QuickTableViewController` as specified in the README. I would still say that IGListKit may still be better for large data sources however.

File: Example-iOS/ViewControllers/DynamicViewController.swift [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          self.tableView.insertRows(at: [IndexPath(row: self.dynamicRows.count, section: 0)], with: .automatic)
+        })
+      ] + dynamicRows
+
+      return [
+        Section(title: "Tap Action", rows: rows as! [Row & RowStyle])
+      ]
+    }
+    set {} // swiftlint:disable:this unused_setter_value
+  }
```
[the remark this answers is not in the document]
Comment (twodayslate) [reply Example-iOS/ViewControllers/DynamicViewController.swift]: The calculations of `tableContents` wouldn't be cached tho. Add a sleep(1) to the `tableContents` getter.

[the remark this answers is not in the document]
Comment (bcylin) [reply Example-iOS/ViewControllers/DynamicViewController.swift]: If you need cache due to the dynamic contents, please implement the `cachedTableContents` in the subclass and override `tableView` in the `DynamicViewController`, e.g. 9dec58a.

[the remark this answers is not in the document]
Comment (twodayslate) [reply Example-iOS/ViewControllers/DynamicViewController.swift]: That works for me. I assume you have no desire to actually put QuickTableView and QuickTableViewDelegate into the package?

[the remark this answers is not in the document]
Comment (bcylin) [reply Example-iOS/ViewControllers/DynamicViewController.swift]: The dynamic table view controller in the example would be nice. 👍

File: Example-iOS/ViewControllers/DynamicViewController.swift [hunk 3]
```diff
@@ -0,0 +1,80 @@
+//
+//  AppearanceViewController.swift
+//  Example-iOS
+//
+//  Created by Ben on 30/01/2018.
```
Comment (bcylin) [inline Example-iOS/ViewControllers/DynamicViewController.swift]:
```suggestion
//  Created by Zac on 30/01/2018.
```

File: Example-iOS/ViewControllers/QuickTableView.swift [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+//
+
+import Foundation
+import UIKit
+
```
Comment (bcylin) [inline Example-iOS/ViewControllers/QuickTableView.swift]: I think the protocol can be moved here above the class.

```suggestion
protocol QuickTableViewDelegate: class
```

Comment (twodayslate) [reply Example-iOS/ViewControllers/QuickTableView.swift]: Should QuickTableView.swift be moved to DynamicViewController.swift as well (to not pollute the ViewControllers folder)?

Comment (bcylin) [reply Example-iOS/ViewControllers/QuickTableView.swift]: It should be all right to have the table view subclass in a different file.

**Comment by twodayslate** [comment]
Thank you for taking the time to review this. Your feedback was appreciated.

**Pull request merged**